core/eolearn/core/eodata_io.py

```diff
@@ -249,9 +269,11 @@ def load(self):
             return self.loaded_value
 
         with self.filesystem.openbin(self.path, 'r') as file_handle:
-            if self.path.endswith(FileFormat.GZIP.extension()):
+            gzip_extension = FileFormat.GZIP.extension()
+            if self.path.endswith(gzip_extension):
+                path = self.path[:-len(gzip_extension)]
```

Comment:
consider `os.path.splitext`, might be cleaner

Reply:
Nice find! Although I'll use `fs.path.splitext` as we are working with OS-agnostic paths.